include/xtypes/DynamicData.hpp

```diff
@@ -235,6 +235,173 @@ class ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate ReadableDynamicDataRef
+    class Iterator
+    {
+    public:
+        Iterator(
+                const ReadableDynamicDataRef& ref)
+            : Iterator(ref, false)
+        {}
+
+        Iterator& operator = (
+                const Iterator& other)
+        {
+            data_ = other.data_;
+            index_ = other.index_;
+            return *this;
+        }
+
+        bool operator == (
+                const Iterator& other) const
+        {
+            return other.data_ == data_ && other.index_ == index_;
+        }
+
+        bool operator != (
+                const Iterator& other) const
+        {
+            return other.data_ != data_ || other.index_ != index_;
+        }
+
+        ReadableDynamicDataRef operator * () const
+        {
+            const CollectionType& collection = static_cast<const CollectionType&>(type_);
+            return ReadableDynamicDataRef(collection.content_type(), collection.get_instance_at(data_, index_));
+        }
+
+        Iterator& operator ++ ()
+        {
+            ++index_;
+            return *this;
+        }
+
+        Iterator operator ++ (
+                int)
+        {
+            Iterator prev = *this;
+            ++index_;
+            return prev;
+        }
+
+    protected:
+
+        friend class ReadableDynamicDataRef;
+
+        Iterator(
+                const ReadableDynamicDataRef& ref,
+                bool end)
+            : type_(ref.type_)
+            , data_(ref.instance_)
+            , index_(end ? ref.size() : 0)
+        {
+        }
+
+        const DynamicType& type_;
+        uint8_t* data_;
```

Comment:
Could you name it as `instance`? In order to only named `data` a `DynamicData` and `instance` when you use a pointer that belongs to a future DynamicData (in order to be consistent with the rest of the library that uses this two names for each case)